diff --git a/Aufgabe_3/src/laden.hh b/Aufgabe_3/src/laden.hh
index 07aa500..88e5853 100644
--- a/Aufgabe_3/src/laden.hh
+++ b/Aufgabe_3/src/laden.hh
@@ -48,6 +48,17 @@ public:
    */
   Regal(string name, Lager &lager, std::set<int> warengruppen);
 
+  /**
+   * @brief Template-Funktion zum Abrufen von Produkten im Regal
+   *
+   * Diese Funktion ruft Produkte aus dem Regal ab und gibt die Ergebnisse an den angegebenen Ausgabeiterator aus.
+   *
+   * @param out Der Ausgabeiterator, der das Ziel faer die Produkte
+   * repraesentiert.
+   */
+  template <class OutputIterator> 
+  void products(OutputIterator out) const;
+
   /**
    * @brief Gibt eine Referenz auf das Lager zurueck, zu dem das Regal gehoert.
    *
